eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -51,7 +45,7 @@ import scala.util.Random
  * Sender for a multi-part payment (see https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#basic-multi-part-payments).
  * The payment will be split into multiple sub-payments that will be sent in parallel.
  */
-class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, relayer: ActorRef, router: ActorRef, register: ActorRef) extends FSMDiagnosticActorLogging[MultiPartPaymentLifecycle.State, MultiPartPaymentLifecycle.Data] {
+class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: ActorRef, register: ActorRef) extends FSMDiagnosticActorLogging[MultiPartPaymentLifecycle.State, MultiPartPaymentLifecycle.Data] {
```

Comment:
Yes but there's a subtlety there because of trampoline (that probably needs to be better documented). The `SendPaymentConfig` contains the final recipient (from the Bolt 11 invoice). When we're using trampoline, the `SendMultiPartPayment` does *not* contain the final recipient, but rather the trampoline node (which is a final recipient of the first part of the route).

We need both to record accurate information about the payment in the DB and in events, which is why they are separate. This trampoline subtlety is quite hard to express easily...